site-service/src/main/java/uk/ac/ox/ndph/mts/site_service/service/SiteServiceImpl.java

```diff
@@ -76,4 +89,60 @@ public String save(Site site) {
         return sites;
     }
 
+    private ValidationResponse validateParentSiteExists(final String parentSiteId) {
+        if (this.siteStore.findById(parentSiteId).isPresent()) {
+            return new ValidationResponse(true, "");
+        }
+        return new ValidationResponse(false, Services.PARENT_NOT_FOUND.message());
+    }
+
+    /**
+     * Find Site By Site Name
+     *
+     * @param siteName the Site to search.
+     * @return site The Site being searched, or none() if not found
+     */
+    public Optional<Site> findSiteByName(String siteName) {
```

Comment:
I don't see this method used by the controller. Maybe keep it package-private for now?

Reply:
Done